icu4c/source/test/fuzzer/converter_fuzzer.cc

```diff
@@ -20,25 +20,34 @@ using deleted_unique_ptr = std::unique_ptr<T, std::function<void(T*)>>;
 
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   UErrorCode status = U_ZERO_ERROR;
-  auto rng = CreateRng(data, size);
-  icu::UnicodeString str(UnicodeStringFromUtf8(data, size));
+  uint16_t rnd = 0;
 
-  const char* converter_name =
-      ucnv_getAvailableName(rng() % ucnv_countAvailable());
+  if (size < 2) {
+    return 0;
+  }
+
+  rnd = *(reinterpret_cast<const uint16_t *>(data));
+  data = data + 2;
+  size = size - 2;
+
+  size_t unistr_size = size/2;
+  std::unique_ptr<char16_t[]> fuzzbuff(new char16_t[unistr_size]);
+  std::memcpy(fuzzbuff.get(), data, unistr_size * 2);
+  
+  icu::UnicodeString fuzzstr(fuzzbuff.get(), unistr_size);
```

Comment:
icu::UnicodeString fuzzstr(false, fuzzbuff.get(), unistr_size);

This is the "Aliasing" constructor - it will cause the UnicodeString to use your fuzzbuff directly, rather than copying the string data to another buffer. If you let UnicodeString use its own buffering, the buffer might be bigger than needed, which could hide buffer overrun errors.

Reply:
Switched constructor to the aliasing type.